libs/execution/src/lib/types/valuetypes/internal-representation-parsing.ts

```diff
@@ -11,10 +11,7 @@ import {
   ValuetypeVisitor,
 } from '@jvalue/jayvee-language-server';
 
-const DECIMAL_COMMA_SEPARATOR_REGEX = /^[+-]?([0-9]*[,])?[0-9]+$/;
-const DECIMAL_DOT_SEPARATOR_REGEX = /^[+-]?([0-9]*[.])?[0-9]+$/;
-
-const INTEGER_REGEX = /^[+-]?[0-9]+$/;
+const NUMBER_REGEX = /^[+-]?([0-9]*[,\\.])?[0-9]+([eE][+\\-]?\d+)?$/;
```

Comment:
I'm curious; what does the `[,\\.]` do?
I only know this syntax here to indicate a logical "or": `[,|.]`.
But I assume that is not what you want to express?

Reply:
To my knowledge the logical or exists, so `,|\.` would be `,` or `.`. With the bracers, you define a group of characters that satisfy the regex, so `[,\.]` is "one of , or ." which is functionally identical to `,|\.`. The difference starts with things like `[0-9]` that you could write as `0|1|2|3|4|5|6|7|8|9`.

But since we have used `[+-]` before (and use `[eE]` later), I think sticking to these two character groups makes sense as well. Even though, seeing this I am not sure why I introduced the double escape in the second `[+-]`.

Reply:
Wouldn't then that sub-expression mean "one of the following: , or  \ or any character" since the backslash is escaped instead of the fullstop?

Reply:
Yeah, I changed this again to not use the escaping character. I think (from the regex testers I used), the escape works as desired but it is just not needed in this case.